dask_ml/metrics/pairwise.py

```diff
@@ -87,7 +104,9 @@ def euclidean_distances(
     return distances if squared else da.sqrt(distances)
 
 
-def check_pairwise_arrays(X, Y, precomputed=False):
+def check_pairwise_arrays(
+    X: ArrayLike, Y: ArrayLike, precomputed: bool = False
+) -> Tuple[ArrayLike, ArrayLike]:
```

Comment:
I don't know typing well enough, but I think there's a way to say that the `Type[X]` will exactly match the type of the first element of the result tuple? So you know that if you put in a dask.Array you'll get back a dask.Array rather than an numpy.ndarray?

Reply:
I am not sure about this :/ .

Reply:
OK to punt on it for now.